### DISCUSSION TITLE
AD material properties vs regular material properties

### DISCUSSION THREAD

**Discussion by rh201** [idaholab/moose#17786] [Q&A Modules: Solid mechanics] [answered]
Hi all, when I run the code which contains block [[Modules/TensorMechanics/Master] [all]], there is an error as follows, do you ever occur the same/similar error and how to correct it?  And what is the difference between AD material properties and regular material properties? Thank you.

_*** ERROR ***
The requested AD material property mechanical_strain is declared as a regular material property. Either retrieve it as a regular material property with getMaterialProperty or declare it as an AD property wtih declareADProperty_

**Comment by vanwdani** [comment, marked as answer]
I don't know exactly what your input file/code looks like, but I would guess you have enabled the AD system in some fashion.  More about the AD: https://mooseframework.inl.gov/bison/automatic_differentiation/index.html

**Reply by rh201** [reply]
Thank you.
